src/lib/Libutil/avltree.c

```diff
@@ -141,6 +159,23 @@ avl_init_tls(void)
 	}
 }
 
+/**
+ * @brief
+ *	return an unique thread index for each new thread
+ *
+ */
+short
+get_thread_index(void)
```

Comment:
Will pthread_getthreadid_np() not work?

Reply:
I recall pthread_getthreadid_np() returns a posix id for the thread. What i need is an integer to index into an array, hence using an internal number

Reply:
Looks like pthread_getthreadid_np() does return an integer, but it's non-posix, which is what the 'np' stands for. So nevermind.

Reply:
yes np is for that, but also the integer is its own number (i meant that when i said posix, i should have said internal number) which i cannot use to index into my array